### ISSUE TITLE
Proposal for changing the time of the first scrape

### ISSUE THREAD

**Issue by FStefanni** [prometheus/prometheus#9251]
Hi,

this is a proposal for a small but useful change.
I am using Prometheus v2.29.1.

This is the current situation:
* If I se a `scrape_interval` of 30m
    * In the `Targets` tab I will se if the connection succeeds only after 30m
    * In the `Graph` tab I will be not able to see the metrics for 30m

I am suggesting to slightly change this behavior, by immediately trying to perform a first scrape.
This will allow:
* To debug immediately possible connection issues
* To inspect immediately the metrics.
* To immediately plot graphs in Grafana

In other words, this speed up the deploy/configuration of the environment, as well as verifying possible 
configuration changes.

Regards

**Comment by roidelapluie** [comment]
Hello,

Prometheus performs scrapes at pre-determined periods. We will not change that because our Query Engine, PromQL, relies on this. It is important to have evenly spaced scrapes across restarts.

Also, Prometheus does not support 30-minutes scrape interval.

**Comment by FStefanni** [comment]
Hi,

> Prometheus performs scrapes at pre-determined periods. We will not change that because our Query Engine, PromQL, relies on this. It is important to have evenly spaced scrapes across restarts.

Sorry, maybe I was not clear (or I am not fully understanding your answer).

I am proposing to just run the first scrape immediately, and then at each equally spaced period.
For an example, if we set a scrape interval of 30m, and we are at 5:00, then
* Current behavior:
   * First scrape at 5:30
   * Then each half, i.e. at 6:00, 6:30 etc.
* Proposed behavior:
   * First scrape immediately, at 5:00
   * Then each half, i.e. at 5:30, 6:00, 6:30 etc.

So the proposed behavior preserves the equally spaced scrape.

Regards

**Comment by roidelapluie** [comment]
We should not wait.

If prometheus starts at 4:59, we would scrape at 5:00.

**Comment by roidelapluie** [comment]
To be more clear:

If prometheus scrapes at 5:00, 5:30, 6:00..
and you restart Prometheus at 5:01 **or** 5:15 **or** 5:20, next scrape will be a 5:30.

**Comment by FStefanni** [comment]
Hi,

ok understood, now it is clear, thank you. 
So I believe there is no an easy way to implement the proposed change, since it would required quite a refactor of the underlying engine.

Regards.
